rclpy/rclpy/node.py

```diff
@@ -1461,12 +1461,18 @@ def destroy_node(self) -> bool:
         # It will be destroyed with other publishers below.
         self._parameter_event_publisher = None
 
-        self.__publishers.clear()
-        self.__subscriptions.clear()
-        self.__clients.clear()
-        self.__services.clear()
-        self.__timers.clear()
-        self.__guards.clear()
+        while self.__publishers:
+            self.destroy_publisher(self.__publishers.pop())
```

Comment:
This is effectively the same as `self.__publishers.clear()`, but the items are removed one at a time. `pop()` removes the publisher from `self.__publishers`, which causes this logic to get skipped.

https://github.com/ros2/rclpy/blob/e566f3e15094dbb065fda72ee5969e826ff76e52/rclpy/rclpy/node.py#L1347-L1355

Reply:
Hah. So would the right thing to do be

```python3
self.destroy_publisher(self.__publishers[0])
```
?

Which would keep destroying the first publisher until there are none more?

Reply:
I think that would do it.

While you're there, I recommend adding a comment above this block that this is a workaround for an unknown bug that causes a deadlock.